include/targets/cuda/shared_memory_cache_helper.h

```diff
@@ -84,12 +100,22 @@ namespace quda
       }
     };
 
-    template <bool dynamic_shared> __device__ __host__ inline std::enable_if_t<dynamic_shared, atom_t *> cache()
+    template <bool dynamic_shared> __device__ __host__ inline std::enable_if_t<dynamic_shared, atom_t*> cache()
```

Comment:
Can this have `const` applied to it (i.e., `[...] cache() const`)? Considering the object returned can be modified but it doesn't actually modify anything within this class. Or, alternatively, is the function below all that's needed?

The reason I ask is so we can potentially close this issue: https://github.com/lattice/quda/issues/1341

Reply:
4334841b9ced537246a6d2a4565ae283ac291055